yb-voyager/cmd/analyzeSchema.go

```diff
@@ -744,22 +745,42 @@ func reportXMLAndXIDDatatype(createTableNode *pg_query.Node_CreateStmt, sqlStmtI
 		*/
 		if column.GetColumnDef() != nil {
 			typeName := ""
-			if len(column.GetColumnDef().GetTypeName().GetNames()) > 0 {
-				typeName = column.GetColumnDef().GetTypeName().GetNames()[0].GetString_().Sval // 0th index as for these non-native types pg_catalog won't be present on first location
+			typeNames := column.GetColumnDef().GetTypeName().GetNames()
+			if len(typeNames) > 0 {
+				typeName = column.GetColumnDef().GetTypeName().GetNames()[len(typeNames)-1].GetString_().Sval // 0th index as for these non-native types pg_catalog won't be present on first location
```

Comment:
comment is probably not valid anymore? we are pickking up the last name in typenames.

Do add a comment explaining why we're picking the last

Reply:
right, will fix